pkg/metrics/sketch_series.go

```diff
@@ -78,6 +79,117 @@ func (sl SketchSeriesList) MarshalJSON() ([]byte, error) {
 	return reqBody.Bytes(), err
 }
 
+// MarshalSplitCompress uses the stream compressor to marshal and compress sketch series payloads.
+// If a compressed payload is larger than the max, a new payload will be generated. This method returns a slice of
+// compressed protobuf marshaled gogen.SketchPayload objects. gogen.SketchPayload is not directly marshaled - instead
+// it's contents are marshaled individually, packed with the appropriate protobuf metadata, and compressed in stream.
+// The resulting payloads (when decompressed) are binary equal to the result of marshaling the whole object at once.
+func (sl SketchSeriesList) MarshalSplitCompress() ([]*[]byte, error) {
+	input := bytes.NewBuffer(make([]byte, 0, 1024))
```

Comment:
Could we have a package var/static for the `1024` size ? Also why 1024 ? it's the max payload size 2MB ?

Reply:
These are the same buffer sizes used in payload builder. 
I ran benchmarks with larger/smaller values and there is not a meaningful difference.

Reply:
OK, I miss the extend part. I just wanted to make sure we were not limiting ourselves to 1MB transaction when we can sends 2MB to the backend.

Reply:
for extra clarity, the compressor constructor pulls in the maximum payload size from the `datadog.yaml` config :)